tests/test_project.py

```diff
@@ -1827,6 +1827,58 @@ def check_root(root=None):
         finally:
             os.chdir(cwd)
 
+    def test_get_job_valid_workspace(self):
+        # Test case: The root-path is the job workspace path.
+        root = self._tmp_dir.name
+        project = signac.init_project(name='testproject', root=root)
+        job = project.open_job({'a': 1})
+        job.init()
+        with job:
+            # The context manager enters the working directory of the job
+            self.assertEqual(project.get_job(), job)
+            self.assertEqual(signac.get_job(), job)
+
+    def test_get_job_invalid_workspace(self):
+        # Test case: The root-path is not the job workspace path.
+        root = self._tmp_dir.name
+        project = signac.init_project(name='testproject', root=root)
+        job = project.open_job({'a': 1})
+        job.init()
+        # We shouldn't be able to find a job while in the workspace directory,
+        # since no signac_statepoint.json exists.
+        cwd = os.getcwd()
+        try:
+            os.chdir(project.workspace())
+            with self.assertRaises(LookupError):
+                project.get_job()
+            with self.assertRaises(LookupError):
+                signac.get_job()
+        finally:
+            os.chdir(cwd)
+
+    def test_get_job_nested_project(self):
+        # Test case: The job workspace dir is also a project root dir.
+        root = self._tmp_dir.name
+        project = signac.init_project(name='testproject', root=root)
+        job = project.open_job({'a': 1})
+        job.init()
+        with job:
+            nestedproject = signac.init_project('nestedproject')
+            nestedproject.open_job({'b': 2}).init()
+            with self.assertRaises(KeyError):
+                project.get_job()
+            with self.assertRaises(KeyError):
+                signac.get_job()
+
+    def test_get_job_subdir_of_project(self):
+        # Test case: The job workspace dir is a sub-directory of the project root dir.
+        pass
+
+    def test_get_job_not_subdir_of_project(self):
```

Comment:
I think I understand this unit test that still needs to implemented, but I forgot what the one above is supposed to test. @bdice Do you remember?

Reply:
Two (three?) tests are unimplemented:
```python
    def test_get_job_subdir_of_project(self):
        # Test case: The job workspace dir is a sub-directory of the project root dir.
        pass
```
This description doesn't make sense to me. I would like to test the case where we have a project with a job, and the current working directory is a subdirectory of the job workspace.
```python
    def test_get_job_not_subdir_of_project(self):
        # Test case: The job workspace dir is not a sub-directory of project root dir.
        # Test case: The job is not actually part of the project that we identified for this job.
        pass
```
I think the first case would be like making a symlink to `my_project/workspace/abc123` and placing that symlink outside of the `my_project` directory. I expect this to fail.

I think the second case is similar to the first, but placing that symlink in an unrelated project `other_project` directory. This should also fail unless that job symlink is placed in the workspace, `other_project/workspace`, in which case it is actually "in the `other_project`." Not sure what to test there.

If you want me to implement these, ping me.